Reading List/Reading List/BookDetailViewController.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  BookDetailViewController.swift
+//  Reading List
+//
+//  Created by Keri Levesque on 1/28/20.
+//  Copyright © 2020 Lambda School. All rights reserved.
+//
+
+import UIKit
+
+class BookDetailViewController: UIViewController {
+
+    @IBOutlet weak var textField: UITextField!
+  
+    @IBOutlet weak var textView: UITextView!
+   
+    var bookController: BookController?
+    var book: Book?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        updateViews()
+
+        // Do any additional setup after loading the view.
+        
+    }
+    
+    func updateViews() {
+           if (book == nil) {
+               self.title = "Add a new book"
+           } else {
+               self.title = book?.title
+               textField.text = book?.title
+               textView.text = book?.reasonToRead
+           }
+       }
+
+    
+    
+    @IBAction func saveButton(_ sender: Any) {
+        if let book = book {
+                guard let titleReady = textField.text, let reasonReady = textView.text else { return }
+                bookController?.editBook(book: book, title: titleReady, reasonToRead: reasonReady)
+                        } else {
+                guard let titleReady = textField.text, let reasonReady = textView.text else { return }
+                bookController?.createBook(named: titleReady, withReason: reasonReady)
+                }
```

Comment:
This is good and works, but would be cleaner if you did you unwrapping first and then did your if let statements.